base/libgit2/callbacks.jl

```diff
@@ -224,7 +229,7 @@ function credentials_callback(libgit2credptr::Ptr{Ptr{Void}}, url_ptr::Cstring,
 
     if isset(allowed_types, Cuint(Consts.CREDTYPE_USERPASS_PLAINTEXT))
         defaultcreds = reset!(UserPasswordCredentials(true), -1)
-        credid = "$schema$host"
+        credid = "$(isempty(schema) ? "ssh" : schema)://$host"
```

Comment:
Note that using the scp-like syntax `git@github.com:JuliaLang/julia.git` or `ssh://git@github.com/JuliaLang/julia.git` both refer to the same git repo using the SSH protocol. 

The advantage of defaulting to using SSH here is that both of these syntaxes will refer to the same `credid`.

See https://git-scm.com/book/id/v2/Git-on-the-Server-The-Protocols (under "The SSH Protocol" section)